trio/tests/test_exports.py

```diff
@@ -18,3 +25,25 @@ def test_core_is_properly_reexported():
                 found += 1
         print(symbol, found)
         assert found == 1
+
+
+def test_pylint_sees_all_non_underscore_symbols_in_namespace():
+    # Test pylints ast to contain the same content as dir(trio)
+    linter = PyLinter()
+    ast_set = set(linter.get_ast(trio.__file__, 'trio'))
+    trio_set = set([symbol for symbol in dir(trio) if symbol[0] != '_'])
+    trio_set.remove('tests')
+    assert trio_set - ast_set == set([])
+
+
+@pytest.mark.skipif(
+    sys.version_info.minor > 7 and sys.version_info.releaselevel != 'final',
+    reason="jedi 3.8-dev not supported"
+)
+def test_jedi_sees_all_completions():
+    # Test the jedi completion library get all in dir(trio)
+    script = jedi.Script(path=trio.__file__)
```

Comment:
I think this should be `script = jedi.Script("import trio; trio.")`

Right now you're telling jedi: "hey, if I had `trio/__init__.py` open in my editor, and went to the last line (or something like that?), and hit TAB, what completions should be displayed?" Which is a bit strange. My snippet says "hey, if I typed `import trio; trio.` in my editor and then hit TAB, what should be displayed?" Which seems like what we want :-)